solr/solr-ref-guide/src/reindexing.adoc

```diff
@@ -0,0 +1,185 @@
+plan to reindex your documents to avoid unintended behavior.
+
+== Upgrades
+
+When upgrading between major versions (for example, from a 7.x release to 8.0 or 8.x), a best practice
+is to always reindex your data.
+The reason for this is that subtle changes may occur in default field type definitions or the underlying code.
+
+[NOTE]
+If you have *not* changed your schema as part of an upgrade from one minor release to another (such as, from 7.x
+to 7.x), you can often get away with reindexing your documents without first deleting your index. However, when
+upgrading to a major release, you should plan to reindex your documents because of the likelihood of changes that break back-compatibility.
+
+== Reindexing Strategies
+
+There are a few approaches available to perform the reindex.
+
+The strategies described below ensure that the Lucene index is completely dropped so you can recreate it to accommodate your changes.
+They allow you to recreate the Lucene index without having Lucene segments lingering with stale data.
+
+=== Delete All Documents
+
+The best approach is to first delete everything from the index, and then index your data again.
+You can delete all documents with a "delete-by-query", such as this:
+
+[source,bash]
+curl -X POST -H 'Content-Type: application/json' --data-binary '{"delete":{"query":"*:*" }}' http://localhost:8983/solr/my_collection/update
+
+It's important to verify that *all* documents have been deleted, as that ensures the Lucene index segments have been
+deleted as well.
+
+To verify that there are no segments in your index, look in the data directory and confirm it is empty.
+Since the data directory can be customized, see the section <<datadir-and-directoryfactory-in-solrconfig.adoc#specifying-a-location-for-index-data-with-the-datadir-parameter,Specifying a Location for Index Data with the dataDir Parameter>>
+for where to look to find the index files.
+
+Note you will need to verify the indexes have been removed in every shard and every replica on every node of a cluster.
+
+Once the indexes have been cleared, you can start reindexing by re-running the original index process.
+
+=== Index to Another Collection
+
+In cases where you cannot take a production collection offline to delete all the documents, one option is to use Solr's <<collections-api.adoc#createalias,collection alias>> feature.
+
+This option is only available for Solr installations running in SolrCloud mode.
+
+With this approach, you will index your documents into a newly created collection and once everything is completed,
+create an alias for the collection and point your front-end at the collection alias. Queries will be routed
+to the new collection seamlessly.
+
+Here is an example of creating an alias that points to a single collection:
+
+[source,bash]
+http://localhost:8983/solr/admin/collections?action=CREATEALIAS&name=myData&collections=oldCollection
+
```

Comment:
Should it say than after that it's safe to delete the old collection using DELETE collections API? I suspect some people may be afraid of the DELETE acting on the alias and being explicit in the doc will save them some googling time

Reply:
Good point, will add.